torch/_dynamo/backends/debugging.py

```diff
@@ -61,12 +70,13 @@ def eager_noexcept(gm, fake_tensor_inputs, **kwargs):
     # This backend is intended to check that dynamo-generated GraphModules
     # do not cause errors.
     def inner(*args):
-        try:
-            return gm(*args)
-        except Exception as e:
-            raise torch._dynamo.exc.TorchDynamoException(
-                "Unexpected exception when running generated GraphModule"
-            ) from e
+        with torch._C.DisableTorchFunctionSubclass():
```

Comment:
don't we need to disable modes here as well?

Reply:
Dynamo disables modes at the beginning of compilation (by popping all of them) already

Reply:
We leave subclass on because it might affect the behavior of individual tensors (ex is_nested and other properties)